lang/gjs/src/gtk4/widget.ts

```diff
@@ -1 +1,19 @@
-// TODO:
+import { Astal, ConstructProps } from "../gtk4"
+import astalify, { BindableChild } from "./astalify"
+import GObject from "gi://GObject"
+
+export type WindowProps = ConstructProps<Window, Astal.Window.ConstructorProps>
+export class Window extends astalify(
+    Astal.Window,
+    Astal.Window.name,
```

Comment:
sorry, I don't know how to make named arguments with default parameters properly in typescript.
what's the preferred way of avoiding providing `clsName` but setting the `childrenSetter` function ?
@Aylur  @kotontrion any suggestions?